rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -25,154 +28,202 @@
 
 
 class YAMLStoryReader(StoryReader):
-    """Class that reads the core training data in a YAML format"""
+    """Class that reads Core training data and rule data in YAML format"""
 
     async def read_from_file(self, filename: Text) -> List[StoryStep]:
-        try:
-            yaml_content = rasa.utils.io.read_yaml_file(filename)
-            if not isinstance(yaml_content, dict):
-                common_utils.raise_warning(
-                    f"Failed to read '{filename}'. It should be a Yaml dict."
-                )
-                return []
-
-            stories = yaml_content.get(KEY_STORIES)  # pytype: disable=attribute-error
-            if not stories:
-                return []
+        """Read stories or rules from file.
 
-            self._parse_stories(stories)
-            self._add_current_stories_to_result()
-            return self.story_steps
+        Args:
+            filename: Path to the story/rule file.
 
+        Returns:
+            `StoryStep`s read from `filename`.
+        """
+        try:
+            yaml_content = rasa.utils.io.read_yaml_file(filename)
         except ValueError as e:
             common_utils.raise_warning(
                 f"Failed to read YAML from '{filename}', it will be skipped. Error: {e}"
             )
+            return []
 
-        return []
+        if not isinstance(yaml_content, dict):
+            common_utils.raise_warning(
+                f"Failed to read '{filename}'. It should be a YAML dictionary."
+            )
+            return []
 
-    def _parse_stories(self, stories: List[Dict[Text, Any]]) -> None:
-        for story_item in stories:
-            if not isinstance(story_item, dict):
+        stories = yaml_content.get(KEY_STORIES, [])
+        self._parse_data(stories, is_rule_data=False)
+
+        rules = yaml_content.get(KEY_RULES, [])
+        self._parse_data(rules, is_rule_data=True)
+
+        self._add_current_stories_to_result()
+
+        return self.story_steps
+
+    def _parse_data(self, data: List[Dict[Text, Any]], is_rule_data: bool) -> None:
+        item_title = self._get_item_title(is_rule_data)
+
+        for item in data:
+            if not isinstance(item, dict):
                 common_utils.raise_warning(
-                    f"Unexpected block found in '{self.source_name}': \n"
-                    f"{story_item}\n"
-                    f"Items under the '{KEY_STORIES}' key must be YAML dictionaries. "
-                    f"It will be skipped.",
-                    docs=DOCS_URL_STORIES,
+                    f"Unexpected block found in '{self.source_name}':\n"
+                    f"{item}\nItems under the "
+                    f"'{self._get_plural_item_title(is_rule_data)}' key must be YAML "
```

Comment:
what does the `get_plural` part refer to?

Reply:
It returns `stories` as opposed to `story`. Also `rules` instead of `rule`.

Reply:
optional: is the `plural` in the name actually important? Up to you 👍

Reply:
yes because there is another method `_get_item_title()` from which this needs to be distinguished